backend/worker/worker.go

```diff
@@ -490,27 +490,33 @@ func (w *Worker) PublicWorker(ctx context.Context) {
 		}(i)
 	}
 
-	wg.Add(1)
-	traceFlushSize := 32 * DefaultBatchFlushSize
-	traceBuffer := &KafkaBatchBuffer{
-		messageQueue: make(chan *kafkaqueue.Message, traceFlushSize),
+	traceWorkers := 1
+	if cfg, err := w.PublicResolver.Store.GetSystemConfiguration(ctx); err == nil {
+		traceWorkers = cfg.TraceWorkers
 	}
-	go func(workerId int) {
-		k := KafkaBatchWorker{
-			KafkaQueue: kafkaqueue.New(
-				ctx,
-				kafkaqueue.GetTopic(kafkaqueue.GetTopicOptions{Type: kafkaqueue.TopicTypeTraces}),
-				kafkaqueue.Consumer, &kafkaqueue.ConfigOverride{QueueCapacity: pointy.Int(traceFlushSize)},
-			),
-			Worker:              w,
-			BatchBuffer:         traceBuffer,
-			BatchFlushSize:      traceFlushSize,
-			BatchedFlushTimeout: DefaultBatchedFlushTimeout,
-			Name:                "traces",
+	traceFlushSize := DefaultBatchFlushSize
+	wg.Add(traceWorkers)
+	for i := 0; i < traceWorkers; i++ {
+		traceBuffer := &KafkaBatchBuffer{
```

Comment:
did we want a separate buffer per worker or should this be out of the loop?

Reply:
i think the kafka message receiving is never the bottleneck here, so it doesn't matter if we have 1 goroutine here, or 32 goroutines sharing the buffer (bottleneck is always the single-threaded flush). i'm putting this back in the for loop so that we can set this to ie. 4 goroutines to have 4 concurrent flushes